src/model/blueprint/Page.ts

```diff
@@ -0,0 +1,46 @@
+import { SubjectType } from '@/model/subject/Subject';
+import { FieldType } from '@/model/field/Field';
+import { Blueprint } from '@/model/blueprint/Blueprint';
+
+export interface PageBlueprint extends Blueprint {
+	type: SubjectType.Page;
+	fields: {
+		title: { type: FieldType.Text; selector?: string };
+		date: { type: FieldType.Date; selector?: string };
```

Comment:
I think maybe pages shouldn't have a date, as I think typically pages on a website will not display a date, so the user will not be able to select it.

Because of this, currently it's not possible to proceed (the Continue button is disabled) because the date has not been selected.

I think the ultimate solution for this is to have optional fields, but for now we could just remove the date field from pages.

Reply:
Yes, data needs to be optional. Regular pages like About, Contact may not have a date, but other kind of pages surely will.

Reply:
Currently, the reason blueprints have an `isValid` is because we need to know whether the blueprint is valid so that we can "short-circuit" the blueprint editor when the blueprint is already valid. This means that, when the user clicks "Import blog posts" and the blueprint is already valid, they don't get to see the blueprint editor, and instead are sent directly to the first step of the crawling flow.

I think ultimately blueprints do need to have some sort of isValid field (or maybe ideally a function), because it should not be possible to start crawling with an invalid blueprint.

I think the ideal scenario would be to:

- Continue having the concept of whether a blueprint is valid or not
- Introduce the concept of optional fields

This way we would be able to have the best of both worlds:

- it's possible to know whether a blueprint is valid or not
- it's possible to proceed without needing to define all fields of the blueprint